### PR TITLE
feat: Optional argument to allow variable redirects on fetch

### PR DESCRIPTION
**Pull request by kyle-cochran**

### CODE REVIEW

File: include/OpenSpaceToolkit/IO/IP/TCP/HTTP/Client.hpp [hunk 1]
```diff
@@ -59,7 +59,7 @@ class Client
     /// @param              [in] aDirectory A destination directory
     /// @return             A file
 
-    static File Fetch(const URL& aUrl, const Directory& aDirectory);
+    static File Fetch(const URL& aUrl, const Directory& aDirectory, const Integer& aFollowCount = 0);
```
Comment (vishwa2710) [inline include/OpenSpaceToolkit/IO/IP/TCP/HTTP/Client.hpp:62]: Maybe make this a `Size` so it can't be negative?

**Review by vishwa2710 — APPROVED** [review]
Just a nit, looks good to me! (Gotta format)

File: include/OpenSpaceToolkit/IO/IP/TCP/HTTP/Client.hpp (hunk 1 above)
Comment (kyle-cochran) [reply include/OpenSpaceToolkit/IO/IP/TCP/HTTP/Client.hpp]: Good looks. This is a much better idea.

**Review by Derollez — APPROVED** [review]

File: src/OpenSpaceToolkit/IO/IP/TCP/HTTP/Client.cpp [hunk 2]
```diff
@@ -227,7 +232,7 @@ File Client::Fetch(const URL& aUrl, const Directory& aDirectory)
         curl_easy_setopt(curlPtr, CURLOPT_WRITEFUNCTION, writeDataFunction);
 
         curl_easy_setopt(curlPtr, CURLOPT_FAILONERROR, true);
-        // curl_easy_setopt(curlPtr, CURLOPT_FOLLOWLOCATION, 1L) ;
+        curl_easy_setopt(curlPtr, CURLOPT_FOLLOWLOCATION, aFollowCount);
```
Comment (Derollez) [inline src/OpenSpaceToolkit/IO/IP/TCP/HTTP/Client.cpp:235]: @kyle-cochran I honestly can't remember why this was commented out. If it is covered with tests and passes, I think we should be good

**Review by vishwa2710 — APPROVED** [review]

**Pull request merged**